pyop2/global_kernel.py

```diff
@@ -331,14 +333,68 @@ def code_to_compile(self):
         from pyop2.codegen.rep2loopy import generate
 
         wrapper = generate(self.builder)
+        if self._iterset._extruded:
+            iname = "layer"
+        else:
+            iname = "n"
+
+        has_matrix = any(arg._is_mat for arg in self._args)
+        has_rw = any(arg.access == op2.RW for arg in self._args)
+        is_cplx = any(arg.dtype.name == 'complex128' for arg in self._args)
```

Comment:
And for these I think you want `self.local_kernel.arguments` instead of `self._args`.

Reply:
I think I want `self.arguments`